u-root.go

```diff
@@ -369,6 +369,12 @@ func Main(l ulog.Logger, buildOpts *gbbgolang.BuildOpts) error {
 }
 
 func validateArg(arg string) bool {
+	// Do the simple thing first: stat the path.
+	// This saves incorrect diagnostics when the
+	// path is a perfectly valid relative path.
+	if _, err := os.Stat(arg); err == nil {
+		return true
```

Comment:
So I haven't checked through all the code ultimately using the arg, but there is a good reason to check for this. `go build` treats `cpu/cmds/cpud` and `./cpu/cmds/cpud` differently. If you were, for example, to pass `cmd/build` to u-root, and os.Stat passed, you would *not* be building `./cmd/build`, you would build `$GOROOT/src/cmd/build` from the Go standard library (or at least, that's what it was pre-Go-modules, I'm just writing this comment based on what I used to know).

Before you change this, look into that. It wasn't just added to be frustrating.

Reply:
good point. I wonder if we might do this:
stat it. IF it exists, and it's a rel path, and it does not start with ./, we prepend ./? All I know is the error really does not make sense. Maybe we could just print something more useful? "relative path starting with ." or something? 

It's just super confusing behavior right now, and it bugs me a little that it's saying something is not a rel path when it really is ...

Reply:
one thing: I did verify, and will verify tomorrow to make sure I'm not dreaming, that the output for  the name ./cpu/cmds/cpud and cpu/cmds/cpud is the same. So I'm no longer sure either of what the rules are.

Reply:
re your point about cmd/build, that certainly was true, and may still be; it's why I made the directory cmds, not cmd, in u-root, because it just got too annoying :-)

Reply:
w/o `./` can probably seen as implicit of w/ `./` -- builds the same -- e.g. `go build` and `go build ./` 

It is only used once, and here in `u-root.go`, to validate the globs passed to u-root, as a valid imports. 

``` 
         // Resolve globs into package imports.
         //
         // Currently allowed format:
         //   Paths to Go package directories; e.g. $GOPATH/src/github.com/u-root/u-root/cmds/*
         //   u-root templates; e.g. all, core, minimal (requires uroot-source be valid)
         //   Import paths of u-root commands; e.g. github.com/u-root/u-root/cmds/* (requires uroot-source)
```

Reply:
so, a test:
I made a fakeo thing, cmd/vet/build.go, in u-root, with a syntax error
```
cat cmd/vet/build.go
package main

func main() {
	go away
}

```


```
./u-root core cmd/vet
13:35:40 Disabling CGO for u-root...
13:35:40 Build environment: GOARCH=amd64 GOOS=linux GOROOT=/home/rminnich/go1.16 GOPATH=/home/rminnich/go CGO_ENABLED=0
13:35:40 WARNING: You are not using one of the recommended Go versions (have = go1.18.3, recommended = [go1.17]).
			Some packages may not compile.
			Go to https://golang.org/doc/install to find out how to install a newer version of Go,
			or use https://godoc.org/golang.org/dl/go1.17 to install an additional version of Go.
13:35:40 NOTE: building with the new gobusybox; to get the old behavior check out commit 8b790de
13:35:40 Skipping package "cmds/core/bind": no buildable Go source files in /home/rminnich/go/src/github.com/u-root/u-root/cmds/core/bind
13:35:40 Skipping package "cmds/core/installcommand": no buildable Go source files in /home/rminnich/go/src/github.com/u-root/u-root/cmds/core/installcommand
13:35:40 Skipping package "cmds/core/timeout": no buildable Go source files in /home/rminnich/go/src/github.com/u-root/u-root/cmds/core/timeout
13:35:40 Skipping package "cmds/core/bind": no buildable Go source files in /home/rminnich/go/src/github.com/u-root/u-root/cmds/core/bind
13:35:40 Skipping package "cmds/core/installcommand": no buildable Go source files in /home/rminnich/go/src/github.com/u-root/u-root/cmds/core/installcommand
13:35:40 Skipping package "cmds/core/timeout": no buildable Go source files in /home/rminnich/go/src/github.com/u-root/u-root/cmds/core/timeout
/home/rminnich/go/src/github.com/u-root/u-root/cmd/vet/build.go:4:9: function must be invoked in go statement
13:35:48 Successfully built "/tmp/initramfs.linux_amd64.cpio" (size 12566164).
```

well, hmm, got an error, built anyway?
but in any event, I can't see any evidence that it was grabbing vet from $GO/src/cmd, as it used to do.

I put a syntax error into $GO/src/cmd/vet/main.go, and that did not trip it up.

I also ran it under strace, looked at all file access, and did not see it using $GO/src/cmd/vet/main.go

I suspect this PR is ok, then, in our new world?

The one disturbing thing is it did install vet into the initramfs, but:
```
rminnich@a300:/tmp/x$ bbin/vet
2022/07/19 13:44:02 Invalid busybox command: ["bbin/vet"]
rminnich@a300:/tmp/x$ ls -l bbin/vet
lrwxrwxrwx 1 rminnich rminnich 2 Jul 19 13:43 bbin/vet -> bb
```

Reply:
hm, that's a headscratcher. I suppose it's alright in the new world.